internal/server/util/jwt_service.go

```diff
@@ -0,0 +1,208 @@
+package util
+
+import (
+	"crypto/x509"
+	stdjson "encoding/json" //nolint: depguard
+	"encoding/pem"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type IJwtService interface {
+	JwtPublicKey() interface{}
+	ParseAuthorization(authorization []string) (*JWTClaims, error)
+}
+
+type JWTClaims struct {
+	jwt.RegisteredClaims
+	Oid        string `json:"oid"`
+	OrgName    string `json:"org_name"`
+	OrgOwner   bool   `json:"user_is_org_owner"`
+	FeatureSet string `json:"feature_set"`
+}
+
+type JwtService struct {
+	jwtPublicKey interface{}
+}
+
+type NewJwtServiceOpts struct {
+	JwtPublicKey string
+}
+
+func New(opts NewJwtServiceOpts) (*JwtService, error) {
+	// parsing in case of bad env variable parsing
+	parsedStr := strings.ReplaceAll(opts.JwtPublicKey, "\\n", "\n")
+
+	raw, rest := pem.Decode([]byte(parsedStr))
+	if len(rest) > 0 {
+		return nil, errors.New("rest found, Jwt key invalid")
```

Comment:
The meaning of `rest found` may be clear for those who know what the code is doing, but otherwise it may be a bit obscure. Can we rephrase it more clearly?

Reply:
Done